### ISSUE TITLE
ExplicitCollectionElementAccessMethod: Do not report map?.get("foo")

### ISSUE THREAD

**Issue by xouabita** [arturbosch/detekt#2391]
The `ExplicitCollectionElementAccessMethod` rule seems really useful but it looks like it is also reporting safe map element access:

https://github.com/arturbosch/detekt/blob/master/detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/ExplicitCollectionElementAccessMethodSpec.kt#L34

Is it possible to not report the following ?

```kotlin
var map: Map<String, Int>? = null
map?.get("foo") 
```

Or is there an elegant way to fix the issue with the current ?

This fixes it, but it doesn't seems optimal...
```kotlin
map?.let { it["foo"] }
```

**Comment by BraisGabin** [comment]
This seems like a false positive. And `map?.let { it["foo"] }` is worst than `map?.get("foo")`.

**Comment by schalkms** [comment]
But this access isn’t save.
It still returns null for a key which hasn’t been found in the map.

**Comment by xouabita** [comment]
I am not sure I follow.... This rule is not about `null` safety but about style:

For now, with this rule enabled, we cannot do `map?.get("foo")` and we need to write `map?.let { it["foo"] }` which looks like a false positive

**Comment by schalkms** [comment]
Okay, now I understand. I'll reopen it in a second.
@xouabita do you have resources to tackle this?

**Comment by xouabita** [comment]
Yes, I can have a look

**Comment by schalkms** [comment]
Cool thanks for helping out! 🙂
